crepecake/src/test/java/com/google/cloud/tools/crepecake/image/LayerBuilderTest.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.crepecake.image;
+
+import com.google.cloud.tools.crepecake.blob.Blob;
+import com.google.cloud.tools.crepecake.blob.Blobs;
+import com.google.common.base.Charsets;
+import com.google.common.io.CharStreams;
+import com.google.common.io.Resources;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.zip.GZIPInputStream;
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
+import org.apache.commons.compress.archivers.tar.TarArchiveInputStream;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.mockito.junit.MockitoJUnitRunner;
+
+/** Tests for {@link LayerBuilder}. */
+@RunWith(MockitoJUnitRunner.class)
+public class LayerBuilderTest {
+
+  @Rule public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  @Test
+  public void testBuild() throws URISyntaxException, IOException {
+    Path layerDirectory = Paths.get(Resources.getResource("layer").toURI());
+
+    LayerBuilder layerBuilder = new LayerBuilder();
+
+    // Adds each file in the layer directory to the layer builder.
+    Files.walk(layerDirectory)
+        .filter(path -> !path.equals(layerDirectory))
+        .forEach(
+            path ->
+                layerBuilder.addFile(path.toFile(), layerDirectory.relativize(path).toString()));
```

Comment:
This is a bit strange. Why do we need to pass the file and the string path to addFile. They seem to contain the same information.